composer.json

```diff
@@ -18,41 +18,38 @@
       "role": "Developer"
     }
   ],
-  "minimum-stability": "stable",
-  "prefer-stable": true,
-  "support": {
-    "issues": "https://github.com/widoz/event-listeners-collection/issues"
-  },
-  "autoload": {
-    "classmap": [
-      "src/"
-    ],
-    "psr-4": {
-      "Widoz\\EventListenersCollection\\": "src"
-    }
-  },
-  "autoload-dev": {
-    "psr-4": {
-      "Widoz\\EventListenersCollectionTests\\": "tests/php",
-      "Widoz\\EventListenersCollectionTests\\Functional\\": "tests/php/Functional"
-    }
-  },
   "require": {
     "php": ">=7.1.33",
     "webmozart/assert": "^1.6"
   },
   "require-dev": {
-    "widoz/project-tests-helper": "dev-phpunit-7",
     "brain/monkey": "^2.3",
-    "phpunit/phpunit": "^7",
+    "fzaninotto/faker": "^1.9",
     "inpsyde/php-coding-standards": "^0.13.3",
+    "phpro/grumphp": "^0.16.2",
+    "phpunit/phpunit": "^7",
     "vimeo/psalm": "^3.8",
-    "fzaninotto/faker": "^1.9",
-    "phpro/grumphp": "^0.16.2"
+    "widoz/project-tests-helper": "dev-phpunit-7"
   },
   "config": {
     "platform": {
       "php": "7.1.33"
     }
+  },
+  "autoload": {
+    "psr-4": {
+      "Widoz\\EventListenersCollection\\": "src/"
+    }
+  },
+  "autoload-dev": {
+    "psr-4": {
+      "Widoz\\EventListenersCollectionTests\\": "tests/php/",
+      "Widoz\\EventListenersCollectionTests\\Functional\\": "tests/php/Functional/"
+    }
+  },
+  "minimum-stability": "stable",
+  "prefer-stable": true,
+  "support": {
+    "issues": "https://github.com/widoz/event-listeners-collection/issues"
```

Comment:
I was thinking, on what the logic of normalize the `composer.json` is based? An arbitrary choice or are there any studies which says, this is the best and correct order for the entries?

Reply:
based on the JSON schema of Composer
https://getcomposer.org/schema.json

Reply:
See https://github.com/ergebnis/composer-normalize

Reply:
I don't know sounds arbitrary as choice, why wouldn't be better alphabetically for example? 
Another reason is that we should add a package to run the normalization on every commit because 

1. You have to be sure the `composer.json` is always in the consistent state 
2. You have to prevent other devs from adding things in the wrong place. 

I don't see it useful in CI because I do not want to do changes to the files in CI env.

What about conflicts? There could be commits which move a lot of parts (it's possible may be not often but possible) which will make resolution of conflict difficult and prone to error. 

Honestly I'm not sure I want to introduce that kind of tool. 

Anyway since I do not see forcing in using it I accept the changes a tantum.

Let me know if you have thoughts about what I wrote and thank you for the PR.

Reply:
It is very simple: Composer has a schema https://getcomposer.org/schema.json that tells us everything, composer-normalize can detect non-conforming composer.json with `--dry-run` AND can actually normalize it.

It is up to you how robust this project will be.

Reply:
As far as I understanding validation and normalization are different processes.

Does the normalize act like a fixer not just moving parts because of order?

Reply:
When ran without `--dry-run` it makes composer.json adhere the JSON schema :)

Reply:
Exactly, just a reorder of the entries nothing more. My concern still stay.
It's ok for to add those changes more because of the `classmap` but I would not introduce the normalization of composer as a commit hook or something that is triggered on every commit / push / pull request.

Reply:
normalization should not run on every commit,
checking for compliance with the schema should